src/App.tsx

```diff
@@ -1,15 +1,34 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const filteredMovies = moviesFromServer.filter(movie => {
+    const {
+      title,
+      description,
+    } = movie;
+
+    const searchRegex = new RegExp(query.trim(), 'gi');
+    const isMatch = searchRegex.test(title) || searchRegex.test(description);
+
+    return isMatch;
+  });
```

Comment:
Let's move out creating `searchRegex` variable to avoid re-creating every iteration in filter

```suggestion
  const searchRegex = new RegExp(query.trim(), 'gi');
  
  const filteredMovies = moviesFromServer.filter(movie => {
    const {
      title,
      description,
    } = movie;
    
    const isMatch = searchRegex.test(title) || searchRegex.test(description);

    return isMatch;
  });
```